pkg/security/agent/telemetry.go

```diff
@@ -56,15 +73,78 @@ func (t *telemetry) run(ctx context.Context, rsa *RuntimeSecurityAgent) {
 			if rsa.storage != nil {
 				rsa.storage.SendTelemetry()
 			}
+		case <-profileCounterTicker.C:
+			if err := t.reportProfiledContainers(); err != nil {
+				log.Debugf("couldn't report profiled containers: %v", err)
+			}
 		}
 	}
 }
 
+type profiledContainer struct {
+	name string
+	tag  string
+}
+
+func (pc *profiledContainer) isValid() bool {
+	return pc.name != "" && pc.tag != ""
+}
+
+func (t *telemetry) fetchConfig() (*api.SecurityConfigMessage, error) {
+	cfg, err := t.runtimeSecurityClient.GetConfig()
+	if err != nil {
+		return cfg, errors.New("couldn't fetch config from runtime security module")
+	}
+	return cfg, nil
+}
+
+func (t *telemetry) reportProfiledContainers() error {
+	cfg, err := t.fetchConfig()
+	if err != nil {
+		return err
+	}
+	if !cfg.ActivityDumpEnabled {
+		return nil
+	}
+
+	profiled := make(map[profiledContainer]bool)
+
+	for _, container := range t.containers.ListRunningContainers() {
+		entry := profiledContainer{
+			name: container.Image.Name,
+			tag:  container.Image.Tag,
+		}
+		if !entry.isValid() {
+			continue
+		}
+		profiled[entry] = false
+	}
+
+	finishProfiling := make([]string, 0)
+	for containerEntry := range t.profiledContainers {
+		profiled[containerEntry] = true
+		finishProfiling = append(finishProfiling, fmt.Sprintf("%s:%s", containerEntry.name, containerEntry.tag))
+	}
+
+	missing := make([]string, 0, len(profiled))
+	for entry, found := range profiled {
+		if !found {
+			missing = append(missing, fmt.Sprintf("%s:%s", entry.name, entry.tag))
+		}
+	}
+
+	if len(missing) > 0 {
+		log.Infof("not yet profiled workloads (%d/%d): %v; finished profiling: %v", len(missing), len(profiled), missing, finishProfiling)
```

Comment:
👋🏻 should we lower the log level here ? I'm afraid Info might be a bit too noisy for customers (specifically on hosts with a huge amount of containers)

Reply:
I think maybe we can keep the info (so that we get it internally) and put this log under a config flag, WDYT ?

Reply:
Sounds good ! 👍🏻

Reply:
done in last push